File: Assets/Typescript~/src/Shared/GroundItem/GroundItemUtil.ts
import { ItemStack } from "Shared/Inventory/ItemStack";
import { TimeUtil } from "Shared/Util/TimeUtil";

export class GroundItemUtil {
	public static CanPickupGroundItem(
		itemStack: ItemStack,
		groundItemNob: NetworkObject,
		characterPosition: Vector3,
	): boolean {
		const dist = characterPosition.sub(groundItemNob.gameObject.transform.position).magnitude;
		if (dist > 1.5) {
			return false;
		}

		const attributes = groundItemNob.gameObject.GetComponent<EasyAttributes>();
		const pickupTime = attributes.GetNumber("pickupTime") ?? 0;
		if (TimeUtil.GetServerTime() < pickupTime) {
			return false;
		}

		return true;
	}
}

File: Assets/Bundles/Shared/Resources/TS/GroundItem/GroundItemUtil.lua
-- Compiled with unity-ts v2.1.0-75
local TimeUtil = require("Shared/TS/Util/TimeUtil").TimeUtil
local GroundItemUtil
do
	GroundItemUtil = setmetatable({}, {
		__tostring = function()
			return "GroundItemUtil"
		end,
	})
	GroundItemUtil.__index = GroundItemUtil
	function GroundItemUtil.new(...)
		local self = setmetatable({}, GroundItemUtil)
		return self:constructor(...) or self
	end
	function GroundItemUtil:constructor()
	end
	function GroundItemUtil:CanPickupGroundItem(itemStack, groundItemNob, characterPosition)
		local _characterPosition = characterPosition
		local _position = groundItemNob.gameObject.transform.position
		local dist = (_characterPosition - _position).magnitude
		if dist > 1.5 then
			return false
		end
		local attributes = groundItemNob.gameObject:GetComponent("EasyAttributes")
		local _condition = attributes:GetNumber("pickupTime")
		if _condition == nil then
			_condition = 0
		end
		local pickupTime = _condition
		if TimeUtil:GetServerTime() < pickupTime then
			return false
		end
		return true
	end
end
return {
	GroundItemUtil = GroundItemUtil,
}
-- ----------------------------------
-- ----------------------------------
